contracts/Marketplace.sol

```diff
@@ -1,74 +1,296 @@
-pragma solidity ^0.4.24;
-
-import "openzeppelin-solidity/contracts/utils/Address.sol";
-import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
-import "openzeppelin-solidity/contracts/lifecycle/Pausable.sol";
-
-contract Service is Ownable, Pausable {
-    bytes32 public sid;
-    bytes32 public latest;
-    mapping (bytes32 => Version) public versions;
-    mapping (bytes32 => mapping(address => bool)) private _versionAccess;
-
-    struct Version {
-        bytes32 id;
-        bytes32 location;
-        uint256 price;
-        bool _exists;
-    }
+pragma solidity >=0.5.0 <0.6.0;
+
+import "../node_modules/openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "../node_modules/openzeppelin-solidity/contracts/lifecycle/Pausable.sol";
+import "../node_modules/openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+
+contract Marketplace is Ownable, Pausable {
+  // ------------------------------------------------------
+  // Structures
+  // ------------------------------------------------------
+
+  struct Service {
+    address owner;
+    bytes sid;
+    uint256 price;
+
+    Version[] versions;
+    Payment[] payments;
```

Comment:
change payments to map?

Reply:
I would like to keep array as main struct for data, but yes adding map for "search"

Reply:
It's ok to keep it as array if we're going to extend payment, otherwise this array is like map and we should use map then.